lang/backend/src/ir/decls.rs

```diff
@@ -0,0 +1,110 @@
+//! High-level intermediate respresentation of the AST after erasure.
+//! This representation is shared between any compiler backends and hence can only make few assumptions about the compilation target.
+
+use codespan::Span;
+use url::Url;
+
+use ast::{Attributes, IdBind, UseDecl, VarBind};
+
+use super::exprs::{Case, Exp};
+use super::kind::Kind;
+
+#[derive(Debug, Clone)]
+pub struct Module {
+    pub uri: Url,
+    pub use_decls: Vec<UseDecl>,
+    pub decls: Vec<Decl>,
+}
+
+#[derive(Debug, Clone)]
+pub enum Decl {
+    Data(Data),
+    Codata(Codata),
+    Def(Def),
+    Codef(Codef),
+    Let(Let),
+}
+
+#[derive(Debug, Clone)]
+pub struct Data {
+    pub span: Option<Span>,
```

Comment:
I would remove source spans from the intermediate representation in order to keep it simpler. I don't think we have a usecase yet for them in the IR.